### DISCUSSION TITLE
Listing all project releases via the API

### DISCUSSION THREAD

**Discussion by runtman** [community/community#24472] [Apps, API and Webhooks]
<p>Hello I am trying to list all releases from a private repo, and I just keep seeing 404 errors.</p>
<p>(Private Repo)</p>
<p><a href="https://github.com/runtman/testy-mctesttest" title="Link to private repo (Just a test one)" rel="nofollow noopener">https://github.com/runtman/testy-mctesttest</a></p>
<p>This has two releases:</p>
<p><img src="https://user-images.githubusercontent.com/6459792/181084809-6754579b-8018-4e59-bacd-8700bd1aaa4c.png" alt="Screenshot from 2019-01-31 19-22-30.png" data-base62-sha1="hYaFVuWFdMXGcgUCY1oxPaZd3Fe" width="458" height="231"></p>
<p>According to this documentation:</p>
<p><a href="https://developer.github.com/v3/repos/releases/" rel="nofollow noopener">https://developer.github.com/v3/repos/releases/</a></p>
<p>It says I need to use the following:</p>
<pre><code>GET /repos/:owner/:repo/releases
</code></pre>
<p>However, when I use this syntax:</p>
<pre><code>curl_setopt($curl,CURLOPT_URL,"https://api.github.com/repos/runtman/testy-mctesttest/releases/")
</code></pre>
<p>I get a 404 returned from the API.</p>
<pre><code>HTTP/1.1 404 Not Found
Server: GitHub.com
Date: Thu, 31 Jan 2019 19:15:48 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 87
Status: 404 Not Found
</code></pre>
<p>What am I doing wrong? I have tried with the id’s instead of names and that didn’t work either, I am definitley authing fine as if I remove that then I get a access denied error instead.</p>

**Comment by nickvanw** [comment, marked as answer]
<p>Hi <a class="mention" href="/runtman">@runtman</a></p>
<p>I believe the answer to this may be as easy as removing the trailing slash from the URL you are trying to get. Using the node Repository as an example, I am able to fetch:</p>
<pre><code>https://api.github.com/repos/nodejs/node/releases
</code></pre>
<p>But, when I try to fetch </p>
<pre><code>https://api.github.com/repos/nodejs/node/releases/
</code></pre>
<p>I get a 404. </p>
<p>Let me know if this works!</p>

**Comment by runtman** [comment]
<p>That’s embararsing, I do get results but not what I would expect:</p>
<pre><code>Array
(
    [url] =&gt; https://api.github.com/repos/runtman/testy-mctesttest/releases
    [content_type] =&gt; application/json; charset=utf-8
    [http_code] =&gt; 200
    [header_size] =&gt; 1208
    [request_size] =&gt; 285
    [filetime] =&gt; -1
    [ssl_verify_result] =&gt; 0
    [redirect_count] =&gt; 0
    [total_time] =&gt; 0.443677
    [namelookup_time] =&gt; 0.012436
    [connect_time] =&gt; 0.105148
    [pretransfer_time] =&gt; 0.312936
    [size_upload] =&gt; 0
    [size_download] =&gt; 5
    [speed_download] =&gt; 11
    [speed_upload] =&gt; 0
    [download_content_length] =&gt; 5
    [upload_content_length] =&gt; -1
    [starttransfer_time] =&gt; 0.44362
    [redirect_time] =&gt; 0
    [redirect_url] =&gt;
    [primary_ip] =&gt; 192.30.253.117
    [certinfo] =&gt; Array
        (
        )

    [primary_port] =&gt; 443
    [local_ip] =&gt; 192.168.1.67
    [local_port] =&gt; 39770
)
</code></pre>
<p>Is this because it’s private? I see no download urls etc </p>

**Comment by runtman** [comment]
<pre><code>HTTP/1.1 200 OK
Server: GitHub.com
Date: Sat, 02 Feb 2019 23:47:04 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 5
Status: 200 OK
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4993
X-RateLimit-Reset: 1549154013
Cache-Control: private, max-age=60, s-maxage=60
Vary: Accept, Authorization, Cookie, X-GitHub-OTP
ETag: "847ca9e05b55c0f672831817ef26025b"
X-OAuth-Scopes: admin:gpg_key, admin:org, admin:org_hook, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user, write:discussion
X-Accepted-OAuth-Scopes: repo
X-GitHub-Media-Type: github.v3; format=json
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
X-GitHub-Request-Id: 9F5C:6489:64DEC0:DD811A:5C562BF7

[

]
</code></pre>
<p>So that’s dumping info and output of the curl request, I don’t see anything useful here</p>

**Comment by runtman** [comment]
<p>I need to get good, thanks :slight_smile:</p>

**Comment by nickvanw** [comment]
<p>Hey <a class="mention" href="/runtman">@runtman</a>!</p>
<p>I believe the issue here is that you’re creating releases via Git Tags, and not actually <a href="https://developer.github.com/v3/repos/releases/#create-a-release" rel="nofollow noopener">creating releases</a>.</p>
<p>If you want to be able to list them via this API, you’ll have to do a bit more! Otherwise, you should be able to use the <a href="https://developer.github.com/v3/repos/#list-tags" rel="nofollow noopener">tag list API</a> to see the information on that page. </p>

**Comment by runtman** [comment]
<p>Aha! Thanks sir.</p>
